modules/server/src/main/scala/higherkindness/compendium/Main.scala

```diff
@@ -18,30 +18,69 @@ package higherkindness.compendium
 
 import cats.effect._
 import cats.syntax.functor._
-import fs2.Stream
+import cats.syntax.flatMap._
+import doobie._
+import doobie.hikari.HikariTransactor
 import higherkindness.compendium.core.{CompendiumService, ProtocolUtils}
-import higherkindness.compendium.db.{DBService, FileDBService}
-import higherkindness.compendium.http.RootService
+import higherkindness.compendium.db.PgDBService
+import higherkindness.compendium.http.{HealthService, RootService}
+import higherkindness.compendium.migrations.Migrations
 import higherkindness.compendium.models.CompendiumConfig
-import higherkindness.compendium.storage.{FileStorage, Storage}
+import higherkindness.compendium.storage.FileStorage
+import org.http4s.server.Router
+import org.http4s.syntax.kleisli._
 import pureconfig.generic.auto._
 
 object Main extends IOApp {
   override def run(args: List[String]): IO[ExitCode] =
-    CompendiumStreamApp.stream[IO].compile.drain.as(ExitCode.Success)
+    CompendiumStreamApp.program[IO]
 }
 
 object CompendiumStreamApp {
 
-  def stream[F[_]: ConcurrentEffect]: Stream[F, ExitCode] =
-    for {
-      conf <- Stream.eval(
-        Effect[F].delay(pureconfig.loadConfigOrThrow[CompendiumConfig]("compendium")))
-      implicit0(storage: Storage[F])                     = FileStorage.impl[F](conf.storage)
-      implicit0(dbService: DBService[F])                 = FileDBService.impl[F]
-      implicit0(utils: ProtocolUtils[F])                 = ProtocolUtils.impl[F]
-      implicit0(compendiumService: CompendiumService[F]) = CompendiumService.impl[F]
-      service                                            = RootService.rootRouteService
-      code <- CompendiumServerStream.serverStream(conf.http, service)
-    } yield code
+  def program[F[_]: ContextShift: ConcurrentEffect: Timer]: F[ExitCode] = {
+
+    Effect[F].delay(pureconfig.loadConfigOrThrow[CompendiumConfig]("compendium")).flatMap {
+      config =>
+        {
+          val transactor: Resource[F, HikariTransactor[F]] =
```

Comment:
Why do not use [`Transactor.fromDriverManager(...)`](https://github.com/tpolecat/doobie/blob/060c63043d04f18bfbf1019a576f4182e369f331/modules/core/src/main/scala/doobie/util/transactor.scala#L346) ?

Reply:
Or you could use also `Stream.resource` to avoid a complete rewrite of `def stream`

Reply:
Well, Hikari is a connection pool well known and used in production. I don't know if the default `Transactor` is good enough. It is up to you anyways.